### PR TITLE
Hardware

### PR DESCRIPTION
**Pull request by gnawinganimal**
Support for motors, with pidf and current limits and all that fancy stuff. Also has support for custom units.

### CODE REVIEW

File: core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+import org.sert2521.sertain.units.*
+import com.ctre.phoenix.motorcontrol.can.TalonSRX as CtreTalon
+import com.ctre.phoenix.motorcontrol.ControlMode as CtreControlMode
+import com.ctre.phoenix.motorcontrol.NeutralMode as CtreNeutralMode
+
+class MotorController<T : MotorId>(
+        val id: T,
+        vararg followerIds: MotorId,
+        val name: String = "ANONYMOUS_MOTOR",
```
Comment (andrewda) [inline core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt:11]: What's name used for? Debug eventually?

**Comment by will-hou** [comment]
It may be nice to add support for SPARK motor controllers in the future as NEOs seem to be getting more popular and we may eventually get some ourselves.

It probably doesn't have to happen now as I'm guessing we won't be getting any NEOs for a while but it'd be good to have on the radar I think. Maybe it could be a good project for some of the incoming members :)

http://www.revrobotics.com/sparkmax-software/

File: core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt (hunk 1 above)
Comment (gnawinganimal) [reply core/src/main/kotlin/org/sert2521/sertain/motors/MotorController.kt:11]: Yeah, debugging. If it turns out to be useless we can remove it.

**Comment by gnawinganimal** [comment]
I agree with supporting neos. I'll make a new pull request for it.

**Pull request merged**